feat(documentation): add icon stories (#978)
swisspost/design-system: packages/documentation/src/stories/foundation/icons/icons.stories.mdx

```bash dark
npm install @swisspost/design-system-icons
```

As they are SVGs, you can include them into your HTML in a few ways depending on how your project is setup.<br/>
Whenever you use the SVGs without our `post-icon` component, we recommend setting the properties `width: 1em` and `height: auto` for easy resizing via `font-size`.

## Working with `post-icon` component

The following passages only work if you use the `post-icon` component.

### Color

By default, icons inherit the current text color of their parent element.<br/>
You can therefore use our [text color utility](/?path=/docs/utilities-text--color) to colorize an icon. Or you can use custom CSS to set the icon color, either via direct style attribute, or via custom classes.

<Canvas withSource="open">
  <Story
    name="Colorizing"
    decorators={[
      Story => (
        <div className="d-flex flex-wrap gap-2" style={{ fontSize: '32px' }}>
          <Story />
        </div>
      ),
    ]}
  >
    <post-icon name="1001" class="text-primary" />
    <post-icon name="1001" class="text-info" />
    <post-icon name="1001" class="text-success" />
    <post-icon name="1001" style={{ color: 'blue' }} />

>     <post-icon name="1001" style={{ color: '#800080' }} />

Author: Where did you found an 8-digit hex color?

>     <post-icon name="1001" style={{ color: '#800080' }} />

Reviewer: Never mind !

        hidden: true,
      },
    },
    viewMode: 'docs',
  }}
/>

# Icons

> <p className="lead">This is our self-made icon library, with 800+ icons so far.</p>

Reviewer 2: ```suggestion
<p className="lead">The official Swiss Post Icon Library for the web with over 800 high-quality scalable vector graphics.</p>
```

> The icons are designed to work with the Swiss Post Design System components. They are SVGs, so they scale quickly and easily and can be styled with CSS. While they are built for us, they will work in any project.<br/>
> Use them together or without the packages `@swisspost/design-system-styles` and `@swisspost/design-system-components`.

Reviewer 2: I don't think we're ready to document usage without the component. Let's focus on that first and extend the documentation when needed.

```suggestion
The icons are designed to work with the Swiss Post Design System components. They are SVGs, so they scale and can be styled with CSS.
```

<!-- this is an invisible, navigation-only story -->

<Story name="Start">
  <div />
</Story>

> ## Usage
> 
> The icons are not automatically installed when using one of the above packages in your project, you must explicitly include them.
> 
> ```bash dark
> npm install @swisspost/design-system-icons
> ```
> 
> As they are SVGs, you can include them into your HTML in a few ways depending on how your project is setup.<br/>
> Whenever you use the SVGs without our `post-icon` component, we recommend setting the properties `width: 1em` and `height: auto` for easy resizing via `font-size`.

Reviewer 2: ```suggestion
## Usage

When using the `post-icon` component, the SVG files are loaded from the [unpkg CDN](https://unpkg.com/) by default. If you like to host the icons yourself, you can follow the steps below.

### Self-hosting the icons
Depending on an external CDN might not be ideal in your case, therefore we provide you with a way of self-hosting the SVG files on your server.

#### Installation
Install the [icons package](https://www.npmjs.com/package/@swisspost/design-system-icons) from npm.

```bash dark
npm install @swisspost/design-system-icons
\```

#### Serving the icons
For your project to be able to access the installed icons, you need a way to copy the icons in a `public` assets folder (this folder might have a different name in your project). The installed SVG files are available under:

`./node_modules/@swisspost/design-system-icons/public/post-icons/[name].svg`

#### Setting the base path
To let the `post-icon` component know where your self-hosted icons are available, set the following meta tag in the `head` section of your page:

```html dark
<html lang="en">
  <head>
    <meta name="design-system-settings" data-post-icon-base="/your/path/to/post-icons/" />
    ...
  </head>

</html>
\```

If you don't have access to your head section, you can also set the base path on each icon:
```html dark
<post-icon name="1001" base="/your/path/to/post-icons/"></post-icon>
\```
```

(Remove backslashes on closing code blocks added for compatibility)

## Working with `post-icon` component

> The following passages only work if you use the `post-icon` component.

Reviewer 2: ```suggestion
The following documentation is using the `post-icon` component.
```
